tests/js/server/utils-tests.js

```diff
@@ -0,0 +1,81 @@
+/*
+
+    Tests for the gpii.express.user.utils component.
+
+ */
+/* eslint-env node */
+"use strict";
+
+var fluid  = require("infusion");
+fluid.logObjectRenderChars = 4096;
+
+var gpii   = fluid.registerNamespace("gpii");
+var jqUnit = require("node-jqunit");
+
+require("../lib/");
+require("../../../src/js/server/lib/datasource");
+
+require("gpii-pouchdb");
+gpii.pouch.loadTestingSupport();
+
+fluid.registerNamespace("gpii.express.user.utils.tests");
+
+fluid.defaults("gpii.express.user.utils.tests.caseHolder", {
+    gradeNames: ["fluid.test.testCaseHolder"],
+    modules: [ {
+        name: "Utils component tests",
+        tests: [{
+            expect: 2,
+            name: "Unlock username/password",
+            sequence: [{
+                funcName: "gpii.express.user.utils.tests.unlockUsernamePassword",
+                args: ["{gpii.express.user.utils}"]
+            }]
+        }]
+    }]
+});
+
+gpii.express.user.utils.tests.unlockUsernamePassword = function (utils) {
+    jqUnit.assert("Ok");
+    // jqUnit.fail("Ok 2");
+    var prom = utils.unlockUser("existing", "password");
+    prom.then(function (data) {
```

Comment:
You have to be very careful when working with QUnit and asynchronous activities.  My gut feel is that this might also be a problem.  In my own work I generally call `jqUnit.stop()` before launching the asynchronous activity, and then call `jqUnit.start()` once either the "then" or "reject" branch is hit.  You could also use the new support for promises in the IoC Test Framework, i.e. a "task" as described here:

https://docs.fluidproject.org/infusion/development/IoCTestingFramework.html#supported-fixture-records